Add enterprise composition and fix frontend acceptance tests

alfrunes opened the pull request.

alfrunes committed:
ci: Update image references for frontend:acceptance tests

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
chore: Document hidden --enterprise flag in frontend/e2e_tests/run

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
chore: Add sensible diffault to env variables in frontend/e2e_tests/run

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

mzedel reviewed (changes requested):
no. Please move the enterprise compose files here as well instead - so that users can evaluate the enterprise solution in a configuration we defined. As a user this change makes the product even less accessible.

alfrunes committed:
refac: Move docker-compose.yml to root and add enterprise override

The enterprise docker composition has been brought as a merge overlay
and is applied by setting
```
COMPOSE_FILE="docker-compose.yml:compose/docker-compose.enterprise.yml"
```
or by adding
`-f docker-compose.yml -f compose/docker-compose.enterprise.yml`
to `docker compose` commands.

NOTE: The enterprise images cannot be built from this repo.

The image references has been updated to consume the upstream images by
default.

Changelog: None
Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
build(make): Align image reference variables with docker compose spec

Replaced MENDER_IMAGE_PREFIX with MENDER_IMAGE_REGISTRY and
MENDER_IMAGE_REPOSITORY.

Changelog: None
Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
docs: Update README.md with instructions on using the docker composition

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
test: Align image references for acceptance test overrides

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
test(workflows): Fix compose file reference in extends attribute

`extends` keyword does not render `include`s so we need to import the
service from the original file (`compose/docker-compose.yml`).

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
test: Refactor e2e run and docker-compose script for enterprise

Removed standalone docker calls to simplify startup/cleanup of
end-to-end tests.
Set the gui-tests-runner scale to 0 by default to avoid getting
SIGKILLed after timing out.
Moved connect-client to enterprise compose file to avoid dangling
containers.

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
ci: Update image references for test:frontend:acceptance*

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
ci: Run test:frontend:acceptance on unprivileged host

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
chore: Dump logs on failure in cleanup hook

Since the shell `-e` option is enabled, the condition for dumping the
logs is never reached. This commit cleans this up and puts it in the
cleanup hook. The SIGINT and SIGTERM does not trigger cleanup to allow
debugging of the running composition.

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
test(e2e): Allow end-to-end tests to run on unprivileged containers

The client identity script relies on privileged access to sysfs.
The connect-client service was removed as it is identical with client.

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
test(gui): Add delay to keystrokes in e2e tests

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
test: Remove assertion about mac address being visible

`mender-inventory-network` script fails on unprivileged hosts since
sysfs is not available.

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
chore(gui): Streamline configuration of e2e test client

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
test(gui): Added `--local` and `--local-visual` convenience flags to e2e

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
ci(gui): Fix publish image references for versioned releases

The published images did not contain the repository path and therefore
contained an invalid reference.

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

alfrunes committed:
ci: Remove job `publish:frontend:enterprise`

The image will be published from the enterprise fork (for now).
Although there's no enterprise pipeline for the frontend repository, the
publish stage relies on the private CI environment.
The publish job for the enterprise pipeline will do nothing but copy the
image from docker hub to our private repository.

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

mzedel reviewed (changes requested):
The last commit is misleading as it implies the enterprise pipeline for the gui image would be any different from what is run here - this should never happen.

mzedel reviewed (approved):
Thank you very much for the changes, the description and the e2e tests adjustments!! it should now be a lot easier to get started with the development 🤞
